ocs_ci/ocs/resources/namespacestore.py

```diff
@@ -184,7 +184,171 @@ def verify_health(self, timeout=60, interval=5):
+            pvc_name (str): Name of the PVC that will host the namespace filesystem
+            pvc_size (int): Size in Gi of the PVC that will host the namespace filesystem
+            sub_path (str): The path to a sub directory inside the PVC FS which the NSS will use as the root directory
+            fs_backend (str): The file system backend type - CEPH_FS | GPFS | NFSv4. Defaults to None.
+
+    """
+    nss_creation_cmd = "namespacestore create "
+    NSS_MAPPING = {
+        constants.AWS_PLATFORM: lambda: (
+            f"aws-s3 {nss_name} "
+            f"--access-key {get_attr_chain(cld_mgr, 'aws_client.access_key')} "
+            f"--secret-key {get_attr_chain(cld_mgr, 'aws_client.secret_key')} "
+            f"--target-bucket {uls_name}"
+        ),
+        constants.AZURE_PLATFORM: lambda: (
+            f"azure-blob {nss_name} "
+            f"--account-key {get_attr_chain(cld_mgr, 'azure_client.credential')} "
+            f"--account-name {get_attr_chain(cld_mgr, 'azure_client.account_name')} "
+            f"--target-blob-container {uls_name}"
+        ),
+        constants.RGW_PLATFORM: lambda: (
+            f"s3-compatible {nss_name} "
+            f"--endpoint {get_attr_chain(cld_mgr, 'rgw_client.endpoint')} "
+            f"--access-key {get_attr_chain(cld_mgr, 'rgw_client.access_key')} "
+            f"--secret-key {get_attr_chain(cld_mgr, 'rgw_client.secret_key')} "
+            f"--target-bucket {uls_name}"
+        ),
+        constants.IBM_COS_PLATFORM: lambda: (
+            f"s3-compatible {nss_name} "
+            f"--endpoint {get_attr_chain(cld_mgr, 'ibmcos_client.endpoint')} "
+            f"--access-key {get_attr_chain(cld_mgr, 'ibmcos_client.access_key')} "
+            f"--secret-key {get_attr_chain(cld_mgr, 'ibmcos_client.secret_key')} "
+            f"--target-bucket {uls_name}"
+        ),
+        constants.NAMESPACE_FILESYSTEM: lambda: (
+            f"nsfs {nss_name} "
+            f"--pvc-name {uls_name} "
+            + (f"--sub-path {nss_tup[2]}" if nss_tup[2] else "")
+            + (f"--fs-backend {nss_tup[3]} " if nss_tup[3] else "")
+        ),
+    }
+    nss_creation_cmd += NSS_MAPPING[platform.lower()]()
+    mcg_obj.exec_mcg_cmd(nss_creation_cmd)
+
+
+def oc_create_namespacestore(
+    nss_name,
+    platform,
+    mcg_obj,
+    uls_name=None,
+    cld_mgr=None,
+    nss_tup=None,
+    nsfs_pvc_name=None,
+):
+    """
+    Create a namespacestore using the MCG CLI
+
+    Args:
+        nss_name (str): Name of the namespacestore
+        platform (str): Platform to create the namespacestore on
+        mcg_obj (MCG): A redundant MCG object, used for uniformity between OC and CLI calls
+        uls_name (str): Name of the ULS bucket to use for the namespacestore
+        cld_mgr (CloudManager): CloudManager object used for supplying the needed connection credentials
+        nss_tup (tuple): A tuple containing the NSFS namespacestore details, in this order:
```

Comment:
I feel this order could error prone. How about converting to dict?

Reply:
This is not possible at the moment; The NSS factory relies on a tuple, and an extensive refactor will be needed to accept anything else. Said refactor is planned to take place in 4.11, but would be too much for 4.10